aten/src/ATen/native/cuda/RNN.cu

```diff
@@ -618,7 +618,7 @@ std::tuple<Tensor, Tensor> _thnn_fused_gru_cell_cuda(
 void checkGRUBackwardSizes(const TensorArg& grad_hy, const TensorArg& workspace) {
   CheckedFrom c = "fused_gru_cell_backward";
   checkDim(c, grad_hy, 2);
-  checkSize(c, workspace, {grad_hy->size(0), grad_hy->size(1) * GRU_WORKSPACE_MULTIPLIER});
+  checkSize(c, workspace, std::vector<int64_t>({grad_hy->size(0), grad_hy->size(1) * GRU_WORKSPACE_MULTIPLIER}));
```

Comment:
An initializer list here should work just as well and be stack allocated

Reply:
with the initializer list I got this error:
```
../aten/src/ATen/native/cuda/RNN.cu(621): error: more than one instance of overloaded function "at::checkSize" matches the argument list:
            function "at::checkSize(at::CheckedFrom, const at::TensorGeometryArg &, at::IntArrayRef)"
../aten/src/ATen/TensorUtils.h(84): here
            function "at::checkSize(at::CheckedFrom, const at::TensorGeometryArg &, c10::SymIntArrayRef)"
../aten/src/ATen/TensorUtils.h(88): here
            argument types are: (at::CheckedFrom, const at::TensorArg, {...})
```

Probably because `int64_t` can implicitly convert to `SymInt`, so... the initializer list can be implicitly converted to both `ArrayRef<int64_t>` and `ArrayRef<SymInt>`?

If this seems like a problematic perf hit, what do you think of using the enum trick to disambiguate the two overloads?

Reply:
`std::initializer_list<c10::SymInt>{blah, blah}`